presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorTableStatisticsProvider.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.spi.connector;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.ConnectorTableLayoutHandle;
+import com.facebook.presto.spi.statistics.TableStatistics;
+
+public interface ConnectorTableStatisticsProvider
+{
+    TableStatistics getTableStatistics(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorTableLayoutHandle layoutHandle);
```

Comment:
> The way TableStatistics uses column names is weird. Instead, this should probably take a list of ColumnHandle and return the column statistics for those columns (in the same position as the original).

It seem to me that having list as a parameter and then list as a return will not be nice to use as you have to do index based mapping between two lists in client code.
Why not just change Map key to `ColumnHandle` from `String`. 

We can add the version `getTableStatistics` which takes additional `Set<ColumnHandle>` if we are afraid that getting statistics for all the columns will imply performance problems. But I would leave it as optional improvement if we actually see problem with simpler SPI.